rr_evgp/src/binary_bayes_filter_obstacle_layer.cpp

```diff
@@ -0,0 +1,250 @@
+        lidar_offset_yaw_ = tf2::getYaw(transform_stamped.transform.rotation);
+
+        last_grid_size_x_ = last_grid_size_y_ = last_origin_x_ = last_origin_y_ = 0;
+        current_ = true;
+    }
+
+    void updateBounds(double robot_x, double robot_y, double robot_yaw, double* min_x, double* min_y, double* max_x,
+                      double* max_y) override {
+        if (!enabled_ || !most_recent_scan_) {
+            return;
+        }
+
+        if (layered_costmap_->isRolling()) {
+            matchSize();
+        }
+
+        lidar_x_ = robot_x + (lidar_offset_x_ * std::cos(robot_yaw) - lidar_offset_y_ * std::sin(robot_yaw));
+        lidar_y_ = robot_y + (lidar_offset_x_ * std::sin(robot_yaw) + lidar_offset_y_ * std::cos(robot_yaw));
+        lidar_yaw_ = robot_yaw + lidar_offset_yaw_;
+
+        scan_ = most_recent_scan_;
+        auto* costmap = layered_costmap_->getCostmap();
+        double resolution = costmap->getResolution();
+
+        double angle = scan_->angle_min;
+        for (double range : scan_->ranges) {
+            if (std::isinf(range)) {
+                range = scan_range_;
+            }
+            double wx = lidar_x_ + range * std::cos(lidar_yaw_ + angle);
+            double wy = lidar_y_ + range * std::sin(lidar_yaw_ + angle);
+            angle += scan_->angle_increment;
+
+            double fudge = 2 * resolution;
+            if (wx - fudge < *min_x)
+                *min_x = std::max(costmap->getOriginX(), wx - fudge);
+            if (wx + fudge > *max_x)
+                *max_x = std::min(costmap->getOriginX() + costmap->getSizeInMetersX(), wx + fudge);
+            if (wy - fudge < *min_y)
+                *min_y = std::max(costmap->getOriginY(), wy - fudge);
+            if (wy + fudge > *max_y)
+                *max_y = std::min(costmap->getOriginY() + costmap->getSizeInMetersY(), wy + fudge);
+        }
+    }
+
+    void updateCosts(costmap_2d::Costmap2D& master_grid, int min_cell_x, int min_cell_y, int max_cell_x,
+                     int max_cell_y) override {
+        if (!enabled_ || !scan_) {
+            return;
+        }
+
+        const double resolution = master_grid.getResolution();
+        const double world_min_x = master_grid.getOriginX();
+        const double world_max_x = world_min_x + master_grid.getSizeInMetersX();
+        const double world_min_y = master_grid.getOriginY();
+        const double world_max_y = world_min_y + master_grid.getSizeInMetersY();
+
+        for (int mx = min_cell_x; mx < max_cell_x; mx++) {
+            for (int my = min_cell_y; my < max_cell_y; my++) {
+                updates_[master_grid.getIndex(mx, my)] = -1;
+            }
+        }
+
+        double angle = scan_->angle_min;
+        for (double raw_range : scan_->ranges) {
+            bool is_hit = (!std::isinf(raw_range) && raw_range <= scan_range_);
+            double range = is_hit ? raw_range : scan_range_;
+            double trace_range = range + (resolution * 0.7);
```

Comment:
Why did you choose to multiply resolution by .7?

Reply:
When the localization drifts around, treating an obstacle as something with a bit of depth (so marking more than 1 cell for a single lidar point) generates nicer-looking maps. I just pushed an update to make it more clear in the code what is going on with this.